src/components/VarLoader.tsx

```diff
@@ -48,79 +53,97 @@ export default class VarLoader extends React.Component<
-        let found = false;
-        this.state.loadedVariables.forEach((loadedVariable: Variable) => {
-          if (newVariable.name == loadedVariable.name) {
-            //newVariables.push(loadedVariable);
-            found = true;
-          }
-        });
-        if (!found) {
-          newVariables.push(newVariable);
+  isSelected(varName: string) {
+    return this.state.selectedVariables.indexOf(varName) >= 0;
+  }
+
+  async loadSelectedVariables() {
+    // Once the load button is clicked, load only the files that were selected.
+    try {
+      this.state.fileVariables.forEach(async (variable: Variable) => {
+        let ind = this.state.selectedVariables.indexOf(variable.cdmsID);
+        if (ind >= 0) {
+          console.log(`Added: ${variable.name}`);
+          await this.props.loadFileVariable(variable);
+        } else {
+          console.log(`Skipped: ${variable.cdmsID}`);
         }
       });
-      this.setState({
-        variables: newVariables
-      });
-    } else {
-      this.setState({
-        variables: variables
-      });
+      // Select only the max number of slabs allowed for plot injection
+      let selection = this.state.selectedVariables;
+      if (selection.length > MAX_SLABS) {
+        selection = selection.slice(0, MAX_SLABS);
+      }
+      // Update the main widget's current selected variables
+      await this.props.updateSelectedVariables(selection);
+      // Reset the selected files in the var loader when done
+      this.setState({ selectedVariables: new Array<string>() });
+    } catch (error) {
+      console.log(error);
     }
   }
 
   /**
    *
    * @param variable The Variable the user has selected to get loaded
    */
-  selectVariableForLoad(variable: Variable) {
+  selectVariableForLoad(variableName: string): void {
+    console.log(`Variable ${variableName} selected and added`);
     this.setState({
-      selectedVariables: this.state.selectedVariables.concat([variable])
+      selectedVariables: this.state.selectedVariables.concat([variableName])
     });
   }
 
   /**
    *
    * @param variable Remove a variable from the list to be loaded
    */
-  deselectVariableForLoad(variable: Variable) {
-    let selectedVars = this.state.selectedVariables.slice();
-    selectedVars.splice(this.state.selectedVariables.indexOf(variable), 1);
+  deselectVariableForLoad(variableName: string) {
+    console.log(`Variable ${variableName} deselected`);
+    let ind: number = this.state.selectedVariables.indexOf(variableName);
+    let selectedVars: Array<string> = this.state.selectedVariables;
+    if (ind >= 0) {
+      selectedVars.splice(ind, 1);
+      console.log("Removed from selected");
+    }
     this.setState({
       selectedVariables: selectedVars
     });
   }
 
+  // Returns true if the variable name has already been loaded into vcdat
+  isLoaded(variableName: string): boolean {
+    if (this.state.unloadedVariables.indexOf(variableName) < 0) {
```

Comment:
refactor this to the one liner.

Reply:
Yup, saw that and fixed before you told me lol